tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/StandaloneTaskMutationAdapter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.tasklist.webapp.service;
+
+import io.camunda.webapps.schema.entities.tasklist.TaskEntity;
+import io.camunda.zeebe.client.ZeebeClient;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("standalone")
+public class StandaloneTaskMutationAdapter implements TaskMutationAdapter {
+
+  private final ZeebeClient zeebeClient;
+
+  public StandaloneTaskMutationAdapter(
+      final @Qualifier("tasklistZeebeClient") ZeebeClient zeebeClient) {
+    this.zeebeClient = zeebeClient;
+  }
+
+  @Override
+  public void assignUserTask(final TaskEntity task, final String assignee) {
+    final var taskKey = task.getKey();
+    zeebeClient.newUserTaskAssignCommand(taskKey).assignee(assignee).send().join();
```

Comment:
@romansmirnov starting from 8.7, this Zeebe client will be always required to be authenticated with client credentials, otherwise it will be blocked by the new [WebSecurityConfig](https://github.com/camunda/camunda/blob/main/authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java). This is not the case in SaaS for example.

Reply:
But how is this different to 8.6?

With 8.6, users can also enable authentication in the Gateway (gRPC and REST), which requires the Zeebe Client to authenticate with client credentials.

With 8.7 by default, authentication will be disabled for *development* purposes, i.e., clients don't need to authenticate. (I wonder if it might be the other way around in the HELM Charts.) However, if authentication is enabled, the Zeebe Client must authenticate with client credentials.

The approach I am following is to differentiate between the old architecture and the new architecture. In the old architecture (in which Tasklist is started via StandaloneTasklist) the communication continues with the Gateway via gRPC/REST - nothing changes. The same applies to authentication. In the new architecture (in which Tasklist runs as part of the Single Application), it will communicate via the broker internally in the C8 cluster. Users or clients only need to authenticate using the v1 REST API to complete (or assign) a task, but Zeebe Client is not used in the new architecture therefore.

Reply:
in 8.6, users can also enable authentication in the Gateway with `identity-auth` profile. This profile is currently used only in SM **and not in SaaS**. in SM, tasklist uses a client id and secret to call Zeebe. These are set by customers or generated by helm.

In 8.7, we will have two new profiles:
- `basic-auth` that I guess will be used for local deployments
- `oidc-auth` that will be used in SaaS and SM. 
- Both, `basic-auth` and `oidc-auth` require authentication.

So it means that in SaaS, starting from 8.7, we will need to define a client id and secret to be used in Zeebe authentication flows, and inject it in Tasklist webapp deployment.

Maybe this something I need to discuss with identity team and it is out of the scope of the initial problem reported in https://github.com/camunda/camunda/issues/21818. The issue will definitely be solved by the changes you made in this pull request 👍